src/interpreter/index.ts

```diff
@@ -0,0 +1,84 @@
+import { programToSpine } from "../common/Spine";
+import {
+  type Node,
+  block,
+  functionCall,
+  isBuiltinIdent,
+  isOfKind,
+} from "../IR";
+import { readsFromInput } from "../common/symbols";
+import { PolygolfError } from "../common/errors";
+import { compilationOptions, compileVariant } from "../common/compile";
+import javascriptLanguage from "../languages/javascript";
+import { required } from "../common/Language";
+import { addVarDeclarations } from "../plugins/block";
+
+const javascriptForInterpreting = {
+  ...javascriptLanguage,
+  phases: [
+    ...javascriptLanguage.phases,
+    required(addVarDeclarations, {
+      name: "instrumentInNogolf",
+      visit(node, spine, context) {
+        if (context.options.level === "nogolf") {
```

Comment:
Why does it need to check for `"nogolf"`?

Reply:
It doesn't, that's a leftover from when I was modifying javascript directly. Good catch.